src/vehicle.cpp

```diff
@@ -5860,39 +5866,25 @@ int vehicle::break_off( int p, int dmg )
 
 bool vehicle::explode_fuel( int p, damage_type type )
 {
-    const itype_id &ft = part_info(p).fuel_type;
-    struct fuel_explosion {
-        // TODO: Move the values below to jsons
-        int explosion_chance_hot ;
-        int explosion_chance_cold;
-        float explosion_factor;
-        bool fiery_explosion;
-        float fuel_size_factor;
-    };
-
-    static const std::map<itype_id, fuel_explosion> explosive_fuels = {{
-        { fuel_type_gasoline,   { 2, 5, 1.0f, true, 0.1f } },
-        { fuel_type_diesel,     { 20, 1000, 0.2f, false, 0.1f } }
-    }};
-
-    const auto iter = explosive_fuels.find( ft );
-    if( iter == explosive_fuels.end() ) {
-        // Not on the list means not explosive
+    const itype_id &ft = part_info( p ).fuel_type;
+    item fuel = item( ft );
+    if( !fuel.has_explosion_data() ) {
         return false;
     }
+    fuel_explosion data = fuel.get_explosion_data();
 
-    const fuel_explosion &data = iter->second;
-    const int pow = 120 * (1 - exp(data.explosion_factor / -5000 * (parts[p].ammo_remaining() * data.fuel_size_factor)));
+    const int pow = 120 * ( 1 - exp( data.explosion_factor / -5000 * ( parts[p].ammo_remaining() *
```

Comment:
`ammo_remaining()` is relatively expensive to calculate (several set inclusion tests), and `exp(...)` is also a bit expensive. Move to inside the `if( one_in(...) )` block's scope as that is the only use of the calculate value `pow` outside of the commented-out debugmsg.

Reply:
that's not really my code, but sure, let's clean it up while I'm here.

Reply:
Thinking about, the compiler should be smart enough to reorder it, and even if doesn't, any out of order processor is automatically going to reorder it. But it's easy enough to change even if I expect it won't have much effect.

Reply:
Also, fixed.